### PR TITLE
GET /api/v1/favorites/:id - Ready for Staging

### PR DESCRIPTION
**Pull request by hsmitha26**
This PR does the following:
* Adds ` /api/v1/favorites/:id` endpoint
* Finds a favorite song by id that is passed in
* If id does not exist in database, responds with status code 404
* If successful, responds with status code 200 and JSON object with song details
* Adds passing tests for happy and sad paths
* All tests are passing

* Adds tear down to POST request test

### CODE REVIEW

File: knexfile.js [hunk 1]
```diff
@@ -23,7 +23,8 @@ module.exports = {
   },
   production: {
     client: 'pg',
-    connection: 'postgres://mztouxgmsntjsh:5908998545484fb1a334ac1123311ecbded76f60212e6fe658de5fa154e3ce4c@ec2-174-129-255-76.compute-1.amazonaws.com:5432/d29t0b8jgphgn',
+    connection: process.env.PRODUCTION_DB,
```
Comment (alect47) [inline knexfile.js]: You might want to put this in `${ }`, otherwise I am not sure it will be a string

File: knexfile.js [hunk 2]
```diff
@@ -34,7 +35,8 @@ module.exports = {
   },
   staging: {
     client: 'pg',
-    connection: 'postgres://jbtgbgdwjucbaj:6200234cfea52892ab933ebd5d4de153976f8d0e758ec762211a921fb9718d67@ec2-174-129-255-46.compute-1.amazonaws.com:5432/d469hfan6hj3ut',
+    connection: process.env.STAGING_DB,
```
Comment (alect47) [inline knexfile.js]: Same as above

File: routes/api/v1/favorites.js [hunk 3]
```diff
@@ -27,4 +27,22 @@ router.post('/', (request, response) => {
     })
 });
 
+router.get('/:id', (request, response) => {
+  let songId = request.params.id
+  database('favorites').where('id', songId).first()
+    .then(songRecord => {
+      if (songRecord) {
+        database('favorites')
```
Comment (alect47) [inline routes/api/v1/favorites.js]: This might be a good place to refactor later to split into a helper method

File: knexfile.js (hunk 1 above)
Comment (hsmitha26) [reply knexfile.js]: Thank you for catching that, I have corrected this.

File: knexfile.js (hunk 2 above)
Comment (hsmitha26) [reply knexfile.js]: Thank you for catching that, I have corrected this.

File: routes/api/v1/favorites.js (hunk 3 above)
Comment (hsmitha26) [reply routes/api/v1/favorites.js]: Agreed!

**Pull request merged**